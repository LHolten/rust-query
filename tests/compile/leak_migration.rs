use rust_query::{
    LocalClient,
    migration::{Config, EasyMigratable, schema},
};

#[schema]
#[version(0..=1)]
enum Schema {
    MyTable {
        #[version(1..)]
        col: i64,
    },
}

fn migrate<'t>(client: &mut LocalClient) {
    let m = client.migrator(Config::open_in_memory()).unwrap();

    let mut sneaky = None;
    m.migrate(|_, _| v1::update::Schema {
        my_table: v1::MyTable::migrate(|prev| {
            sneaky = Some(prev);
            todo!()
        }),
    });
}

fn main() {}
