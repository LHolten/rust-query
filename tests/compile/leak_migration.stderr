error[E0521]: borrowed data escapes outside of closure
  --> tests/compile/leak_migration.rs:21:13
   |
18 |     let mut sneaky = None;
   |         ---------- `sneaky` declared here, outside of the closure body
19 |     m.migrate(|_, _| v1::update::Schema {
20 |         my_table: v1::MyTable::migrate(|prev| {
   |                                         ---- `prev` is a reference that is only valid in the closure body
21 |             sneaky = Some(prev);
   |             ^^^^^^ `prev` escapes the closure body here
